ksz8863/examples/simple_switch/main/simple_switch_main.c

```diff
@@ -179,37 +181,28 @@ static void got_ip_event_handler(void *arg, esp_event_base_t event_base,
     ESP_LOGI(TAG, "~~~~~~~~~~~");
 }
 
-esp_err_t i2c_init(i2c_port_t i2c_master_port, i2c_config_t *i2c_conf)
-{
-    esp_err_t ret;
-
-    ESP_GOTO_ON_ERROR(i2c_param_config(i2c_master_port, i2c_conf), err, TAG, "I2C parameters configuration failed");
-    ESP_GOTO_ON_ERROR(i2c_driver_install(i2c_master_port, i2c_conf->mode, 0, 0, 0), err, TAG, "I2C driver install failed");
-
-    return ESP_OK;
-err:
-    return ret;
-}
-
 // board specific initialization routine, user to update per specific needs
 esp_err_t ksz8863_board_specific_init(esp_eth_handle_t eth_handle)
 {
     esp_err_t ret = ESP_OK;
 
 #if CONFIG_EXAMPLE_CTRL_I2C
     // initialize I2C interface
-    i2c_config_t i2c_bus_config = {
-        .mode = I2C_MODE_MASTER,
-        .sda_io_num = CONFIG_EXAMPLE_I2C_SDA_GPIO,
+    i2c_master_bus_config_t i2c_mst_config = {
+        .clk_source = I2C_CLK_SRC_DEFAULT,
+        .i2c_port = CONFIG_EXAMPLE_I2C_MASTER_PORT,
         .scl_io_num = CONFIG_EXAMPLE_I2C_SCL_GPIO,
-        .sda_pullup_en = GPIO_PULLUP_ENABLE,
-        .scl_pullup_en = GPIO_PULLUP_ENABLE,
-        .master.clk_speed = CONFIG_EXAMPLE_I2C_CLOCK_KHZ * 1000,
+        .sda_io_num = CONFIG_EXAMPLE_I2C_SDA_GPIO,
+        .glitch_ignore_cnt = 7,
+        .flags.enable_internal_pullup = true
```

Comment:
No need to be set to true in my opinion. It's common practice the I2C bus to be always designed with external pullups.

Reply:
Please fix in all applicable examples.